### PR TITLE
Parse active connection information

### PR DESCRIPTION
**Pull request by chenj47**
Several key features involved in this pull request:
1. Create the proto buff file that defines the ActiveConn (Active Connection) Information.
2. Update the script to generate the pb.go files.
3. Implement the basic function to parse active connection information and store the results using proto buff. Also, add several test cases for the active connection part.

Note that the usage of proto buff is just a tryout in this PR since I want to see if proto buff can be easily used to send data to the front end. If it does not work well, the information will be stored in CSV then.

### CODE REVIEW

File: regen_proto.sh [hunk 1]
```diff
@@ -19,8 +19,9 @@
 # To regenerate the compiled Go files if the protos are modified.
 cd $GOPATH/src/
 
-protoc --go_out=. github.com/googleinterns/sensor-historian/pb/build_proto/*.proto
-protoc --go_out=. github.com/googleinterns/sensor-historian/pb/batterystats_proto/*.proto
-protoc --go_out=. github.com/googleinterns/sensor-historian/pb/metrics_proto/*.proto
-protoc --go_out=. github.com/googleinterns/sensor-historian/pb/usagestats_proto/*.proto
-protoc --go_out=. github.com/googleinterns/sensor-historian/pb/session_proto/*.proto
\ No newline at end of file
+protoc -I=github.com/googleinterns/sensor-historian/pb/activeconnection_proto/ --go_out=. github.com/googleinterns/sensor-historian/pb/activeconnection_proto/*.proto
```
Comment (agarwal-v) [inline regen_proto.sh]: Is -I flag needed?

File: sensorserviceutils/sensorserviceutils.go [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+	"time"
+
+	"github.com/googleinterns/sensor-historian/bugreportutils"
+	"github.com/googleinterns/sensor-historian/csv"
+	"github.com/googleinterns/sensor-historian/historianutils"
+	acpb "github.com/googleinterns/sensor-historian/pb/activeconnection_proto"
+)
+
+var (
+	// Each of the connectionLineXRE is a regular expression to match the X line of
```
Comment (agarwal-v) [inline sensorserviceutils/sensorserviceutils.go]: connLineXRE

File: sensorserviceutils/sensorserviceutils.go [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+	buf      *bytes.Buffer
+	csvState *csv.State
+	errs     []error
+
+	// sensors is a map from sensor number to sensorInfo, it directly adoptes the Sensors map
+	// define in the bugreportutils packaged.
+	sensors map[int32]bugreportutils.SensorInfo
+	// activeConns is a map from the active connection number to the corresponding connection info.
+	activeConns map[int32]*acpb.ActiveConn
+	// activeSensors is a map from sensor number to
```
Comment (agarwal-v) [inline sensorserviceutils/sensorserviceutils.go]: Reword comment

File: sensorserviceutils/sensorserviceutils.go [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+	// a map from packageName to the active connection number.
+	activeSensors map[int32]map[string]int32
+	// historyByUID is a map from a key to SubscriptionInfo.
```
Comment (agarwal-v) [inline sensorserviceutils/sensorserviceutils.go]: history or historybyUID?

File: sensorserviceutils/sensorserviceutils.go [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+	for p.valid() {
+		l := p.line() // Read the current line and advance the line position.
+		// Active connection parsing.
+		if m, _ := historianutils.SubexpNames(activeConnRE, l); m {
+			if err := p.extractActiveConnInfo(); err != nil {
+				p.errs = append(p.errs, err)
+			}
+			continue
+		}
+		// Previous registration parising.
```
Comment (agarwal-v) [inline sensorserviceutils/sensorserviceutils.go]: Remove this code as it is not being usedright now.

**Review by agarwal-v — COMMENTED** [review]
Need to rebase on previous pull request to be able to focus on changes in this request.

**Comment by chenj47** [comment]
Closing this PR to create a new one that better focus on the change for parse-activeConn branch.

**Pull request closed without merging**